src/themis/secure_message_wrapper.c

```diff
@@ -135,21 +135,57 @@ themis_secure_message_verifier_t* themis_secure_message_verifier_init(const uint
     return ctx;
 }
 
+static inline uint64_t total_signed_message_length(const themis_secure_signed_message_hdr_t* msg)
+{
+    /* We're using uint64_t to avoid overflows. Length components are uint32_t. */
+    uint64_t length = sizeof(themis_secure_signed_message_hdr_t);
```

Comment:
i like the trick of using `uint64_t`, but do we care how this code works on x32?

Reply:
AFAIK `uint64_t` should work perfectly on 32-bit systems. Yes, it will take few more asm commands (and CPU cycles) to perform same operation on 64-bit integers, but IMO _this_ func has no issues.
Can somebody prove or disprove my words?

Reply:
> but do we care how this code works on x32?

As long as it produces the same results, I don't care :)

> Can somebody prove or disprove my words?

You absolutely correct. That's the compiler's problem now how to emulate 64-bit arithmetic on a machine with 32-bit registers only. They have been doing this all time since when the CPUs were 8-bit and you had to emulate 16-bit integers for multiplication results to be correct.